Objects/complexobject.c

```diff
@@ -546,12 +548,12 @@ complex_pow(PyObject *v, PyObject *w, PyObject *z)
     // a faster and more accurate algorithm.
     if (b.imag == 0.0 && b.real == floor(b.real) && fabs(b.real) <= 100.0) {
         p = c_powi(a, (long)b.real);
+        _Py_ADJUST_ERANGE2(p.real, p.imag);
```

Comment:
That might be moved to ``c_powi()``, but that will make patch slightly larger.  I think this is fine, as ``c_powi()`` is not a part of C API.